terraform/cloudwatch.tf.json.template.py

```diff
@@ -15,6 +21,17 @@
 )
 
 
+def alarm_resource_name(threshold: MetricThreshold) -> str:
+    # FIXME: Change the parts used in the alarm resource name once the
+    #        '{foo}cachehealth' handlers have been renamed to remove '{foo}'.
+    #        https://github.com/DataBiosphere/azul/issues/5337
+    parts = (
+        threshold.handler_name if threshold.handler_name else threshold.lambda_name,
+        threshold.metric
+    )
+    return '_'.join(parts)
```

Comment:
I think this would make more sense as a method (ideally a property) of the `MetricThreshold` class (i.e. change the argument from `threshold` to `self`).

Reply:
> I think this would make more sense as a method (ideally a property) of the MetricThreshold class

Doing so would put Terraform specific information (the name of the alarm resource) in the chalice module. Part of the goal I was aiming for is keeping the Chalice specifics (e.g. knowledge of `handler_map` and `_original_func`) separate from the Terraform specifics.

Granted, while this isn't completely accomplished (the metric names are needed in `metric_thresholds()` to lookup the threshold values), I think having the Chalice code create the Terraform resource names should be avoided.